posthog/models/app_metrics2/sql.py

```diff
@@ -0,0 +1,215 @@
+from django.conf import settings
+
+from posthog.clickhouse.kafka_engine import KAFKA_COLUMNS_WITH_PARTITION, kafka_engine, ttl_period
+from posthog.clickhouse.table_engines import (
+    AggregatingMergeTree,
+    Distributed,
+    ReplicationScheme,
+)
+from posthog.kafka_client.topics import KAFKA_APP_METRICS2
+
+APP_METRICS2_TTL_DAYS = 90
+
+SHARDED_APP_METRICS2_TABLE_ENGINE = lambda: AggregatingMergeTree(
+    "sharded_app_metrics2", replication_scheme=ReplicationScheme.SHARDED
+)
+
+BASE_APP_METRICS2_COLUMNS = """
+    team_id Int64,
+    timestamp DateTime64(6, 'UTC'),
+    -- The name of the service or product that generated the metrics.
+    -- Examples: plugins, hog
+    app_source LowCardinality(String),
+    -- An id for the app source.
+    -- Set app_source to avoid collision with ids from other app sources if the id generation is not safe.
+    -- Examples: A plugin id, a hog application id
+    app_source_id String,
+    -- A secondary id e.g. for the instance of app_source that generated this metric.
+    -- This may be ommitted if app_source is a singleton.
+    -- Examples: A plugin config id, a hog application config id
+    instance_id String,
+    successes SimpleAggregateFunction(sum, Int64),
+    skipped SimpleAggregateFunction(sum, Int64),
+    failures SimpleAggregateFunction(sum, Int64),
+    error_type String
```

Comment:
I'm wondering if there is any strong reason why not to make this more generic? 

From the work I'm doing so far there is a bunch of metrics that would be really useful to store here
`success | failure | requestSuccess | requestFailure | timeout | filtered | skipped_due_to_disabled` etc.

If we aggregate on the metric name it should still condense nicely I would imagine? 

 ```suggestion
    metric_name LowCardinality(String),
    count SimpleAggregateFunction(sum, Int64)
```

Reply:
Interesting, I can't think of a strong reason.

My best guess at the historical reasons for why it was this way in `app_metrics` (1) is that this table was probably created to support this page: https://us.posthog.com/project/2/pipeline/destinations/457/metrics

Your `metric_name` / `sum` would be able to support the same page with minor query changes, with the relatively minor tradeoff that we'd be going a little "schema-less." It can be hard to know if an arbitrary string is a success or a failure or what. But maybe the future of that page is we just break out each unique type + count and we're done?

Reply:
This seems like a fine idea, but I worry about us always needing to update the UI and alerting - specifically if some sort of a new failure category is added. Specifically spark graph for example only has success and failure, we can't display 7 lines there it would be too noisy.

A potential way to get the best of both worlds is to have top level success / failure - which can be used by alerting and sparkgraph and then subcategories, which can be used in metrics page (or in alerts to ignore too).

something like this:
```
successes SimpleAggregateFunction(sum, Int64),
failures SimpleAggregateFunction(sum, Int64),
category LowCardinality(String),
```

Reply:
I'm :+1: to @benjackwhite's suggestion here.
```
metric_name LowCardinality(String),
count SimpleAggregateFunction(sum, Int64)
```

Is ideal and can represent almost everything that you want now and in the future :heavy_check_mark:

Reply:
I pushed up one like @tiina303 suggested. Rather than a single `count` column, we force you to at least pick one of `successes`, `skipped` or `failures` -- which gives us some way to aggregate / understand counts without having to know about every `metric_name`. Does that seem reasonable @benjackwhite ? It seems like your `success | failure | requestSuccess | requestFailure | timeout | filtered | skipped_due_to_disabled` would all fit into there.

```
    metric_name LowCardinality(String),
    successes SimpleAggregateFunction(sum, Int64),
    skipped SimpleAggregateFunction(sum, Int64),
    failures SimpleAggregateFunction(sum, Int64),
```

And I dropped `error_type` -- non-metric logs / debug into should go to the `log` table, and not in there, as discussed above. (That was an artifact from `app_metrics` (1) that we should remove.)

Reply:
Talked with Ben, went with `metric_kind` and `count` so this can be totally flexible.

i.e. kind could be `error` and name could be `timeout` or `sql_error` or whatever